File: src/java/CdbWebPortal/src/java/gov/anl/aps/cdb/common/exceptions/ConfigurationError.java
/*
 * Copyright (c) UChicago Argonne, LLC. All rights reserved.
 * See LICENSE file.
 */
package gov.anl.aps.cdb.common.exceptions;

import gov.anl.aps.cdb.common.constants.CdbStatus;

/**
 * Configuration error exception.
 */
public class ConfigurationError extends CdbException {

    /**
     * Default constructor.
     */
    public ConfigurationError() {
        super();
    }

    /**
     * Constructor using error message.
     *
     * @param message error message
     */
    public ConfigurationError(String message) {
        super(message);
    }

    /**
     * Constructor sing throwable object.
     *
     * @param throwable throwable object
     */
    public ConfigurationError(Throwable throwable) {
        super(throwable);
    }

    /**
     * Constructor using error message and throwable object.
     *
     * @param message error message
     * @param throwable throwable object
     */
    public ConfigurationError(String message, Throwable throwable) {
        super(message, throwable);
    }

    @Override
    public int getErrorCode() {
        return CdbStatus.CDB_CONFIGURATION_ERROR;
    }    
}

File: src/python/cdb/common/exceptions/configurationError.py
#!/usr/bin/env python

"""
Copyright (c) UChicago Argonne, LLC. All rights reserved.
See LICENSE file.
"""


#
# Configuration error class.
#

#######################################################################

from cdb.common.constants import cdbStatus 
from cdb.common.exceptions.cdbException import CdbException

#######################################################################

class ConfigurationError(CdbException):
    def __init__ (self, error='', **kwargs):
        CdbException.__init__(self, error, cdbStatus.CDB_CONFIGURATION_ERROR, **kwargs)
